### DISCUSSION TITLE
How to send binary using uart.write()

### DISCUSSION THREAD

**Discussion by skysurf76** [micropython/micropython#9786] [Using MicroPython] [answered]
I'm working on communicating with a custom firmware on another device, and I'm trying to use a micropython pyboard to send certain sequences to manipulate the menu on the other system.  Specifically I need to be able to send  escape (ASCII 27) followed by a d.

I tried uart.write(0x1B) and I get the following error..

TypeError: object with buffer protocol required.

Using usart.write('abc') works fine.  I just need a way to insert an escape.

Thanks in advance for any help.

**Comment by skysurf76** [comment, marked as answer]
Got it...

Jus as an example, to send a carriage return you can do uart.write((10).to_bytes(1, 'little'))

where 10 is the decimal code for newline.  The same line works for sending any character via its ASCII code.

**Comment by dlech** [comment]
You can also use `'\x1b'`

**Reply by skysurf76** [reply]
I like your way a lot better.  Much more elegant.  Thanks!
